### ISSUE TITLE
java tensorflow api :  Malformed TF_STRING tensor; too short to hold number of elements 

### ISSUE THREAD

**Issue by astrung** [tensorflow/tensorflow#8834]
First, when i tried LabelImage [example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/src/main/java/org/tensorflow/examples/LabelImage.java) with inception model ver 5, everything was good.

Then i tried with an older inception model ( ver3 ) and i saw that both model have different input and output.

In ver5, our input tensor name is : "Input" , with dtype = FLOAT
In ver 3, our input tensor name is "DecodeJpeg/contents" , with dtype = STRING.

So i change LabelExample example with new name for both input and output : `Tensor result = s.runner().feed("input", string_tensor_image).fetch("output")` >> `s.runner().feed("DecodeJpeg/contents", image).fetch("softmax")` .

Also, i changed Image tensor to STRING type : 

    Tensor float_tensor_image = s.runner().fetch(output.op().name()).run().get(0);
    byte[] bytes = new byte[res.numBytes()];
    ByteBuffer buffer = ByteBuffer.wrap(bytes);
    res.writeTo(buffer);
    Tensor string_tensor_image = Tensor.create(DataType.STRING,res.shape(),buffer);

It looked good when i printed both tensor :
 
    FLOAT tensor with shape [1, 224, 224, 3]
    STRING tensor with shape [1, 224, 224, 3]

But after feeding to the graph, i get this error : 
`Exception in thread "main" java.lang.IllegalArgumentException: Malformed TF_STRING tensor; too short to hold number of elements.`

I have tried everything i can, but no results. How can i fix it ?

**Comment by astrung** [comment]
well, pls help

**Comment by asimshankar** [comment]
This question is probably better posed on [stackoverflow](http://stackoverflow.com/questions/tagged/tensorflow) as we try to keep the github issues focused on bugs and feature requests.

If you do post there, probably also makes sense to provide more information (for example, a link to the exact model you're using if possible)

**Comment by astrung** [comment]
@asimshankar this is model ver 5 : https://storage.googleapis.com/download.tensorflow.org/models/inception5h.zip .

And this is model ver 3 : http://download.tensorflow.org/models/image/imagenet/inception-2015-12-05.tgz

I download both from google. Also, i think it may be a bugs

**Comment by astrung** [comment]
By increasing number of bytes in array, it is solved, but then i get new error :
 
    Tensor float_tensor = s.runner().fetch(output.op().name()).run().get(0);
    byte[] bytes = new byte[float_tensor.numBytes()*64];
    ByteBuffer buffer = ByteBuffer.wrap(bytes);
    res.writeTo(buffer);
    long[] shape = {};
    Tensor string_tensor = Tensor.create(DataType.STRING, shape, buffer);
    return string_tensor;

this is new string tensor : `STRING tensor with shape []`

But i get new error : 

    java.lang.IllegalArgumentException: Invalid JPEG data, size 0
     [[Node: DecodeJpeg = DecodeJpeg[acceptable_fraction=1, channels=3, dct_method="",                               fancy_upscaling=true, ratio=1, try_recover_truncated=false, _device="/job:localhost/replica:0/task:0/cpu:0"]

well, i have no idea about how to fix it ? can someone suggest me ?

**Comment by asimshankar** [comment]
As mentioned earlier, this question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there.

Having more complete code will help there - for example, it isn't clear what `res` and `float_tensor` are in your code and what you're feeding in. The `DecodeJpeg` operation consumes the file contents of a JPEG file as input, is that what you're feeding in? While in the other model, the `input` node consumes the normalized pixel values as floats. 

I strongly suggest you pursue this on stackoverflow as it most likely is an issue with your use of the new model and not a bug in TensorFlow
